arch/xtensa/src/esp32s2/esp32s2_spiram.c

```diff
@@ -0,0 +1,450 @@
+   * ADDRESSS ANYMORE!
+   */
+
+  g_allocable_vaddr_start = vaddr_start;
+  g_allocable_vaddr_end   = vaddr_start + psram_available_size;
+}
+
+/****************************************************************************
+ * Name: esp_spiram_test
+ *
+ * Description:
+ *   Simple RAM test. Writes a word every 32 bytes. Takes about a second
+ *   to complete for 4MiB. Returns true when RAM seems OK, false when test
+ *   fails. WARNING: Do not run this before the 2nd cpu has been initialized
+ *   (in a two-core system) or after the heap allocator has taken ownership
+ *   of the memory.
+ *
+ * Input Parameters:
+ *   None
+ *
+ * Returned Value:
+ *   True on success or False on failure
+ *
+ ****************************************************************************/
+
+bool esp_spiram_test(void)
+{
+  volatile int *spiram = (volatile int *)g_mapped_vaddr_start;
+
+  size_t s = g_mapped_size;
+  size_t p;
+  int errct = 0;
+  int initial_err = -1;
+
+  for (p = 0; p < (s / sizeof(int)); p += 8)
+    {
+      spiram[p] = p ^ 0xaaaaaaaa;
+    }
+
+  for (p = 0; p < (s / sizeof(int)); p += 8)
+    {
+      if (spiram[p] != (p ^ 0xaaaaaaaa))
+        {
+          errct++;
+          if (errct == 1)
+            {
+              initial_err = p * sizeof(int);
+            }
+
+          if (errct < 4)
+            {
+              merr("SPI SRAM error @ %08x:%08x/%08x \n", &spiram[p],
+                   spiram[p], p ^ 0xaaaaaaaa);
+            }
+        }
+    }
+
+  if (errct != 0)
+    {
+      merr("SPI SRAM memory test fail. %d/%d writes failed, first @ %X\n",
+           errct, s / 32, initial_err + SOC_EXTRAM_DATA_LOW);
+      return false;
+    }
+  else
+    {
+      minfo("SPI SRAM memory test OK!");
+      return true;
+    }
+}
+
+uint32_t esp_spiram_instruction_access_enabled(void)
+{
+  return g_instruction_in_spiram;
+}
+
+uint32_t esp_spiram_rodata_access_enabled(void)
+{
+  return g_rodata_in_spiram;
+}
+
+#if defined(CONFIG_ESP32S2_SPIRAM_FETCH_INSTRUCTIONS)
+int esp_spiram_enable_instruction_access(void)
+{
+  /* `pages_for_flash` will be overwritten, however it influences the psram
+   * size to be added to the heap allocator.
+   */
+
+  abort();
+}
+#endif
+
+#if defined(CONFIG_ESP32S2_SPIRAM_RODATA)
+int esp_spiram_enable_rodata_access(void)
+{
+  /* `pages_for_flash` will be overwritten, however it influences the psram
+   * size to be added to the heap allocator.
+   */
+
+  abort();
+}
+#endif
+
+#if defined(CONFIG_ESP32S2_SPIRAM_FETCH_INSTRUCTIONS)
+void instruction_flash_page_info_init(void)
+{
+  uint32_t instr_page_cnt = ((uint32_t)&_instruction_reserved_end -
+                            SOC_IROM_LOW + MMU_PAGE_SIZE - 1) /
+                            MMU_PAGE_SIZE;
+
+  g_instr_start_page = *(volatile uint32_t *)(DR_REG_MMU_TABLE +
+                                              CACHE_IROM_MMU_START);
+  g_instr_start_page &= MMU_ADDRESS_MASK;
+  g_instr_end_page = g_instr_start_page + instr_page_cnt - 1;
+}
```

Comment:
```suggestion
```
These functions are only applicable for **ESP32-S3**, so they should be removed.